src/foam/nanos/ruler/Rule.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.nanos.ruler',
+  name: 'Rule',
+
+  documentation: 'Rule model represents rules(actions) that need to be applied in case passed object satisfies provided predicate.',
+
+  javaImports: [
+    'foam.core.FObject',
+    'foam.core.X'
+  ],
+
+  properties: [
+    {
+      class: 'Long',
+      name: 'id',
+      documentation: 'Sequence number.'
+    },
+    {
+      class: 'String',
+      name: 'name',
+      documentation: 'Rule name for human readability.'
+    },
+    {
+      class: 'Int',
+      name: 'priority',
+      documentation: 'Priority defines the order in which rules are to be applied.'+
+      'Rules with a higher priority are to be applied first.'+
+      'The convention for values is ints that are multiple of 10.'
+    },
+    {
+      class: 'Boolean',
+      name: 'stops',
+      documentation: 'When set to true, the rule prevents execution of the following rules within the same group.'
+    },
+    {
+      class: 'String',
+      name: 'ruleGroup',
+      documentation: 'ruleGroup defines sets of rules related to the same action.'
+    },
+    {
+      class: 'String',
+      name: 'documentation'
+    },
```

Comment:
Maybe add something like this here:
      view: {
        class: 'foam.u2.tag.TextArea',
        rows: 12, cols: 80
      }

Reply:
@kgrgreer added